apps/site/components/Common/Button/index.tsx

```diff
@@ -17,17 +22,46 @@ const Button: FC<ButtonProps> = ({
   href = undefined,
   children,
   className,
+  onClick,
   ...props
-}) => (
-  <Link
-    role="button"
-    href={disabled ? undefined : href}
-    aria-disabled={disabled}
-    className={classNames(styles.button, styles[kind], className)}
-    {...props}
-  >
-    {children}
-  </Link>
-);
+}) => {
+  //to handle the keyboard interactions, specifically for Enter and Space keys
+  const handleKeyDown = (e: KeyboardEvent<HTMLAnchorElement>) => {
+    if (disabled) return;
+
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      if (onClick) {
+        onClick(e as unknown as MouseEvent<HTMLAnchorElement>);
+      }
+    }
+  };
+
+  // to manage mouse click events for the component, providing behavior consistent with the disabled state
+  const handleClick = (e: MouseEvent<HTMLAnchorElement>) => {
+    if (disabled) {
+      e.preventDefault();
+      return;
+    }
+    if (onClick) {
+      onClick(e);
+    }
+  };
+
+  return (
+    <Link
+      role="button"
+      href={disabled ? undefined : href}
+      aria-disabled={disabled}
+      className={classNames(styles.button, styles[kind], className)}
+      tabIndex={disabled ? -1 : 0} // Ensure focusable if not disabled
+      onClick={handleClick}
```

Comment:
IMO instead of disabled on the function call, you should do:

```suggestion
      onClick={disabled || onClick}
```

Or something similar

Reply:
On both the onClick and the onKeyDown probably. Not sure if you need to override the default behaviour, but you could also do:

```jsx
onClick={disabled ? preventDefaultFn : onClickHandler}
```

Reply:
I think keeping the disabled logic in the handleClick function for clarity, maintainability, and accessibility, as it ensures consistent behavior for both mouse and keyboard interactions. onClick={disabled || onClick} risks unintended behavior and lacks flexibility for future enhancements